### ISSUE TITLE
Error message. 

### ISSUE THREAD

**Issue by aymercury** [karl-b/reddit-api#3]
I got this error message when I try to use your gem. Is it because my ruby version is too old (1.9.1)? gem install worked fine.

$ruby random_reddit_oath.rb
WARN: Unresolved specs during Gem::Specification.reset:
      domain_name (~> 0.5)
      netrc (~> 0.7)
WARN: Clearing out unresolved specs.
Please report a bug if this causes problems.
/opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:69:in `require': /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/reddit-api-0.1.2/lib/reddit/internal/connection.rb:25: syntax error, unexpected tLABEL (SyntaxError)
..._agent_title, request_throttle: nil, max_retries: nil)
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/reddit-api-0.1.2/lib/reddit/internal/connection.rb:25: Can't assign to nil
...t_title, request_throttle: nil, max_retries: nil)
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/reddit-api-0.1.2/lib/reddit/internal/connection.rb:143: syntax error, unexpected keyword_end, expecting $end
        from /opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:69:in`require'
        from /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/reddit-api-0.1.2/lib/reddit/api.rb:11:in `<top (required)>'
        from /opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:128:in`require'
        from /opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:128:in `rescue in require'
        from /opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:39:in`require'
        from random_reddit_oath.rb:1:in `<main>'

**Comment by karl-b** [comment]
Hello, 

**tl;dr: Ruby version 2.0+ is required.** 

Im surprised the gem install worked! I just tried to reproduce it really quick and its failing to resolve dependencies for me (fails to find rest-client > 1.0). 

Even if I did get around those though the api has been programed using design patterns only available to ruby in version 2.0+. 

The exception shown we have: 

`"connection.rb:25: syntax error, unexpected tLABEL (SyntaxError)"`

The code on line 25: 

`def initialize(username, password, client_id, secret, user_agent_title, request_throttle: nil, max_retries: nil)`

It is using defaults for named parameters, which was not a feature in 1.9.*. 
http://stackoverflow.com/questions/16322706/optional-arguments-with-default-value-in-ruby

So the short answer is ruby version 1.9 is not supported, It is a quite old version of ruby released about 7 years ago. If there is a specific use case or demand I can take a look at it. 

The simplest option though would be moving to a newer version!

**Comment by aymercury** [comment]
Thanks! I tried two other reddit wrappers, but they refused to install. ;-)
